tests/sharding/test_mainchain_handler.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+
+from viper import compiler
+
+from eth_tester.backends.pyevm.main import get_default_account_keys
+
+from evm.utils.address import generate_contract_address
+
+import eth_utils
+
+from evm.chains.sharding.mainchain_handler.vmc_utils import (
+    decode_contract_call_result,
+    mk_contract_tx_obj,
+)
+
+from evm.chains.sharding.mainchain_handler.mainchain_handler import (
+    MainchainHandler,
+)
+
+test_keys = get_default_account_keys()
+
+@pytest.fixture
+def chain_handler():
+    return MainchainHandler(use_eth_tester=True)
+
+def test_tester_chain_handler(chain_handler):
+    chain_handler.mine(1)
+    code = """
+num_test: public(num)
+
+@public
+def __init__():
+    self.num_test = 42
+
+@public
+def update_num_test(_num_test: num):
+    self.num_test = _num_test
+"""
```

Comment:
I would suggest pulling this block of viper code out to the module level so that it doesn't interupt the indentation of this function body.

Reply:
Done it. Will it even helpful that I replace the code with compiled bytecode, thus we can remove the dependency on viper?